src/renderer/viz/expressions/ramp.js

```diff
@@ -184,16 +185,31 @@ export default class Ramp extends BaseExpression {
             };
         }
 
+        let inline = `texture2D(texRamp${this._uid}, vec2((${input.inline}-keyMin${this._uid})/keyWidth${this._uid}, 0.5))`;
+        if (this.input.type === 'category' && this.input.isA(Property)) {
```

Comment:
`this.input.type === inputTypes.CATEGORY`

Reply:
May I disagree? :joy: I think this is clearer and more concise

Reply:
it was a matter to be consistent with the rest of the file, but we can change this in a future refactor, so no problem 👍